### PR TITLE
MHQ Serialization Cleanup 1: MekHqXmlSerializable and Serializable Removals

### PR DESCRIPTION
**Pull request by Windchild292**
I accidentally mixed two branches, the first a XML mess reduction branch and the second a MekHQ code cleanup branch. Thankfully, the cleanup was whitespace isolated outside of three useless semicolon fixes, so this branch is still pretty much just the serialization changes.

The overall changes within are:
1. Removes MekHqXmlSerializable
2. Removes java.io.Serializable from MekHQ
3. Removes serialVersionUID from MekHQ
4. Moves writeEntityWithCrewToXML to MekHQXMLUtil
5. Updates some MekHqXmlUtil setups to latest
6. Migrates a few MekHQ inline setups to MekHqXmlUtil
7. Fixes a ton of spacing issues across MekHQ
8. Removes the three useless semicolon uses in MekHQ

To Review: Whitespace differences off

### CODE REVIEW

**Comment by Windchild292** [comment]
The LGTM issue is a new detection of a pre-existing issue. I'll handle it alongside the other open LGTM issue once this PR is merged.

**Review by SJuliez — APPROVED** [review]

**Pull request merged**
